### DISCUSSION TITLE
Add prefix or suffix to repeated column names after unnesting recursively

### DISCUSSION THREAD

**Discussion by EvanZ** [duckdb/duckdb#13323] [Ideas]
Granted, this may be a very niche feature, but I have the following use case. I am unnesting basketball play-by-play data which has the key `id` used several times, so that in the end of a recursive unnesting I end up with several `id` columns that I'm not sure how to differentiate with a subsequent select or join. Giving an actual example here:

unnested data:
```
 play = {'id': 401638583102999901, 'sequenceNumber': 102999901, 'type': {'id': 402, 'text': End Game}, 'text': End of Game, 'awayScore': 65, 'homeScore': 85, 'period': {'number': 2, 'displayValue': 2nd Half}, 'clock': {'displayValue': 0:00}, 'scoringPlay': false, 'scoreValue': 0, 'team': NULL, 'wallclock': 2024-03-21 20:13:13, 'shootingPlay': false, 'coordinate': {'x': 50, 'y': -5}, 'participants': NULL}
index = 383
```

After unnesting (using `.mode line`):
```
            id = 401638583102999901
sequenceNumber = 102999901
            id = 402
          text = End Game
          text = End of Game
     awayScore = 65
     homeScore = 85
        number = 2
  displayValue = 2nd Half
  displayValue = 0:00
   scoringPlay = false
    scoreValue = 0
            id = 
     wallclock = 2024-03-21 20:13:13
  shootingPlay = false
             x = 50
             y = -5
  participants = 
         index = 383
```

You can see there are 3 `id` columns, 2 `text` columns, and 2 `displayValue` columns. What would be nice is if I could add a prefix or suffix index to each of these columns to be able to distinguish them in a select statement, for example:

```
            id _1= 401638583102999901
sequenceNumber = 102999901
            id _2= 402
          text_1 = End Game
          text _2= End of Game
     awayScore = 65
     homeScore = 85
        number = 2
  displayValue_1 = 2nd Half
  displayValue_2 = 0:00
   scoringPlay = false
    scoreValue = 0
            id_3 = 
     wallclock = 2024-03-21 20:13:13
  shootingPlay = false
             x = 50
             y = -5
  participants = 
         index = 383
```

**Comment by soerenwolfers** [comment]
Even better if the column names were `id` and `type_id` (possibly gated behind an optional parameter and with a customizable join string)

**Reply by EvanZ** [reply]
Yep great idea.

**Comment by cmdlineluser** [comment]
I did post something similar recently, but there was a lack of feedback:

- https://github.com/duckdb/duckdb/discussions/11012

Basically something equivalent to what `json_normalize` does in pandas/polars
